src/library/coverartcache.cpp

```diff
@@ -173,9 +182,10 @@ void CoverArtCache::coverLoaded() {
         // we have to be sure that res.cover.hash is unique
         // because insert replaces the images with the same key
         QString cacheKey = pixmapCacheKey(
-                res.cover.hash, res.cover.resizedToWidth);
+                res.cover.hash, res.cover.resizedToWidth * res.cover.devicePixelRatio);
```

Comment:
The multiplication could be wrapped into a member function `CoverArt::actualWidth()`